File: 알고리즘/2/How Many Numbers Are Smaller Than the Current Number.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        unordered_map<int, int> map;
        unordered_set<int> s(nums.begin(), nums.end());

        for(auto ele1 : s){
            for(auto ele2 : nums){
                if(ele2 < ele1) map[ele1]++;
            }
        }

        vector<int> answer;

        for(auto ele : nums){
            answer.push_back(map[ele]);
        }
        
        return answer;    
    }
};

File: algorithms/2/How Many Numbers Are Smaller Than the Current Number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        set1 = set(nums)
        dict1 = dict()

        for ele in set1:
            for ele1 in nums:
                dict1[ele1] = dict1.get(ele1, 0) + 1
        
        return [dict1.get(ele) for ele in nums ]
        